### PR TITLE
Text about error messages [I18N-ACTION-1015]

### PR DESCRIPTION
**Pull request by aphillips**
This includes reorganizing the section on localization somewhat. See commit notes for details.

***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/55.html" title="Last updated on Dec 3, 2021, 5:25 PM UTC (ee54c4e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/55/6bb258c...aphillips:ee54c4e.html" title="Last updated on Dec 3, 2021, 5:25 PM UTC (ee54c4e)">Diff</a>

***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/55.html" title="Last updated on Jan 20, 2022, 6:04 PM UTC (a550c0e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/55/6bb258c...aphillips:a550c0e.html" title="Last updated on Jan 20, 2022, 6:04 PM UTC (a550c0e)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2670,33 +2666,51 @@ <h3>Localization</h3>
...
-	<div class="req" id="l10n_api_error_message_language">
+	<div class="req" id="l10n_api_message_language">
 	<p class="advisement">All natural language fields or messages, including error messages, defined by a given API or protocol SHOULD be localized into the preferred locale of the user or, if that language is not available, supplied with a suitable fallback or default.</p>
 	</div>
 
 	<div class="req" id="l10n_api_lang_nego">
 	<p class="advisement">Specifications for APIs or protocols SHOULD define how the user's locale is determined (this is sometimes called <a href="#lang_negotiation">language negotiation</a>).</p>
 	</div>
-
+
```
Comment (xfq) [inline index.html]:
```suggestion

```

File: index.html [hunk 2]
```diff
@@ -2670,33 +2666,51 @@ <h3>Localization</h3>
...
 	<div class="req" id="l10n_api_lang_defaulting">
 	<p class="advisement">Specifications MAY define a specific default language for messages or errors in an API or protocol.</p>
 	</div>
 
 <p class="note">Specifications do not need to require that messages be returned in all possible or all available locales. It is sufficient to make it possible to localize the end-user's customer experience. Implementations can choose which languages or locales to support.</p>
 
+<section id="errors" class="subtopic">
+<h3>Errors and Exceptions</h3>
+
+<p>Protocols, APIs, and document formats sometimes provide a field to pass a human-readable error or exception message from a service to the caller in the form of a string. In general, and as <a href="#l10n_api_message_metadata">indicated above</a>, any natural language text conveying human readable messages or human readable content needs to be associated with language and direction metadata. Where this metadata is missing, the processing or display of the text might be compromised.</p>
```
Comment (xfq) [inline index.html]:
```suggestion
<p>Protocols, APIs, and document formats sometimes provide a field to pass a human-readable error or exception message from a service to the caller in the form of a string. In general, and as <a href="#l10n_api_message_metadata">indicated above</a>, any natural language text conveying human-readable messages or human-readable content needs to be associated with language and direction metadata. Where this metadata is missing, the processing or display of the text might be compromised.</p>
```

File: index.html [hunk 3]
```diff
@@ -2670,33 +2666,51 @@ <h3>Localization</h3>
...
+
+<p>Often the intention of the specification author in providing an error or exception message is to convey debugging information to a software developer. Specification authors sometimes assume that error or exception messages are not seen by end users; that software developers will prefer these messages to be unlocalized or appear in a specific language (usually English); or that there are other "practical reasons" why localization of error messages can turn out to be a barrier. For example, there are annecdotes about developers finding it easier to search the Web with the (usually obscure) text of an error because the message itself is insufficiently good at explaining the problem. Searching for this text might produce a result in the developer's preferred language that is more helpful.</p>
```
Comment (xfq) [inline index.html]: annecdotes -> anecdotes

File: index.html [hunk 4]
```diff
@@ -427,11 +427,11 @@ <h3>Basic requirements</h3>
 
 
 	<div class="req" id="dir_paragraphs">
-	<p class="advisement">It must be possible to indicate base direction for each individual paragraph-level item of natural language text that will be read by someone.</p>
+	<p class="advisement">It must be possible to indicate base direction for each individual paragraph-level item of <a>natural language</a> text that will be read by someone.</p>
```
Comment (aphillips) [inline index.html:430]: @r12a I'd like to remove the awkward ".. that will be read by someone", perhaps by changing 'natural language' here to 'textual content'. Thoughts?

**Comment by xfq** [comment]
I'm confused. Is this PR about error messages or file naming restrictions?

**Comment by xfq** [comment]
Ah, I just saw the email in member-i18n-core. Never mind.

**Comment by aphillips** [comment]
I'm going to merge these changes based on our conversation of 2022-01-20. Note that some text should be reviewed again by the WG and this change depends on changes by Richard to the glossary.

**Pull request merged**